### ISSUE TITLE
OSX desktop client does not allow custom port to be specified

### ISSUE THREAD

**Issue by ran404** [zulip/zulip-desktop#582]
---

- **Operating System**:
  - [ ] Windows
  - [ ] Linux/Ubutnu
  - [x] macOS
- **Clear steps to reproduce the issue**:
On the initial "Add a Zulip organisation" page, enter an url with a custom port, e.g:
https://zulip.example.com:11443

Then click "Connect".
A spinner shows up, the app does not load.

The problem appears to be that the the input form strips away the port number. I'm able to login only after editing ~/Library/Application Support/Zulip/config/domain.json, and manually change the url key from "https://zulip.example.com" to "https://zulip.example.com:11443"

- **Relevant error messages and/or screenshots**:

Port not reflected in the config file

**Comment by akashnimare** [comment]
@abhigyank can you look into it since you have worked on the proxy thing?

**Comment by timabbott** [comment]
This is a high-priority issue, since it prevents some users from getting the app up and working.

**Comment by akashnimare** [comment]
> The problem appears to be that the the input form strips away the port number

I don't think we strip the port since I can add the `https://chat.zulip.org:443`, the problem seems to be somewhere in the request module. Also, can you provide us the zulip server you're trying to connect to so that I can see what's wrong?

**Comment by akashnimare** [comment]
@ran404 can you please provide more info on this?

**Comment by superbiche** [comment]
@akashnimare I can reproduce this issue with a fresh Zulip instance available on the exact same port, on Mac OS Mojave 10.14.1, but not because of a port issue (see #596).
Test instance is available at https://zulip2.superbiche.co:11443 (no org).

I just created this instance with docker-zulip real quick to try and help fix this issue, but there were no errors related to the port. 
To be sure about that, I just added `listen *:11444 ssl http2;` to the server definition for https://zulip.superbiche.co (has 1 org) so that I could try adding https://zulip.superbiche.co:11444, and it worked like a charm (with the fix from #595 indeed).

So I'm pretty sure this is related to #596
